### DISCUSSION TITLE
Using 'hincrby' with byte[] codec for values

### DISCUSSION THREAD

**Discussion by joanbalaguero** [redis/lettuce#2306] [Q&A]
We have a lettuce connection using 'StringAndByteArrayCodec()) which uses a string codec for keys and byte array codec for values. This is because we are storing compressed xml files. But since all fields must be byte[], we are also transforming long,int, ... to byte[].

It works well setting and getting a long value:
````
this.connect().sync().hset("hello", "creationTime", Utils.longToByteArray(25L));
Utils.byteArrayToLong((byte[])this.connect().sync().hget("hello", "creationTime"))); --> prints 25
````
But I don't know how to apply an 'hincrby' to this field:
````
this.connect().sync().hincrby("hello", "creationTime", 1);
````
I get a "io.lettuce.core.RedisCommandExecutionException: ERR hash value is not an integer"

Is there any way to do this?

**Comment by mp911de** [comment]
I have no insights what `longToByteArray` is doing.
